benchexec/runexecutor.py

```diff
@@ -1064,57 +1051,36 @@ def _get_cgroup_measurements(self, cgroups, ru_child, result):
-                    logging.debug(
-                        "Could not read CPU time for core %s from kernel: %s", core, e
-                    )
+            for core, coretime in cgroups.read_usage_per_cpu().items():
+                result[f"cputime-cpu{core}"] = coretime
+
+            cpu_pressure = cgroups.read_cpu_pressure()
+            if cpu_pressure is not None:
+                result["total-cpu-pressure-some"] = cpu_pressure
 
-        if MEMORY in cgroups:
-            # This measurement reads the maximum number of bytes of RAM+Swap the process used.
-            # For more details, c.f. the kernel documentation:
-            # https://www.kernel.org/doc/Documentation/cgroups/memory.txt
-            memUsageFile = "memsw.max_usage_in_bytes"
-            if not cgroups.has_value(MEMORY, memUsageFile):
-                memUsageFile = "max_usage_in_bytes"
-            if not cgroups.has_value(MEMORY, memUsageFile):
-                logging.warning("Memory-usage is not available due to missing files.")
+        if cgroups.MEMORY in cgroups:
+            max_mem_usage = cgroups.read_max_mem_usage()
+            if max_mem_usage is None:
+                logging.warning(
+                    "Memory-usage is not available for cgroups v2 or due to missing files."
+                )
             else:
-                try:
-                    result["memory"] = int(cgroups.get_value(MEMORY, memUsageFile))
-                except OSError as e:
-                    if e.errno == errno.ENOTSUP:
-                        # kernel responds with operation unsupported if this is disabled
-                        logging.critical(
-                            "Kernel does not track swap memory usage, cannot measure memory usage."
-                            " Please set swapaccount=1 on your kernel command line."
-                        )
-                    else:
-                        raise e
-
-        if BLKIO in cgroups:
-            blkio_bytes_file = "throttle.io_service_bytes"
-            if cgroups.has_value(BLKIO, blkio_bytes_file):
-                bytes_read = 0
-                bytes_written = 0
-                for blkio_line in cgroups.get_file_lines(BLKIO, blkio_bytes_file):
-                    try:
-                        dev_no, io_type, bytes_amount = blkio_line.split(" ")
-                        if io_type == "Read":
-                            bytes_read += int(bytes_amount)
-                        elif io_type == "Write":
-                            bytes_written += int(bytes_amount)
-                    except ValueError:
-                        pass  # There are irrelevant lines in this file with a different structure
-                result["blkio-read"] = bytes_read
-                result["blkio-write"] = bytes_written
+                result["memory"] = max_mem_usage
+
+            oom_count = cgroups.read_oom_count()
+            if oom_count:
+                result["oom"] = oom_count
```

Comment:
This is actually the number of killed processes due to OOM, not the OOM count. But it is not really interesting for users, so since f2839b7d5e15c6c0571e6260a877b13a5a9dbffe we do not return it.